controller/tests/handlers/fsadmin_test.py

```diff
@@ -0,0 +1,103 @@
+"""Tests for fsadmin route."""
+
+from __future__ import annotations
+
+import pytest
+from httpx import AsyncClient
+from kubernetes_asyncio.client import V1Pod
+from safir.testing.kubernetes import MockKubernetesApi
+
+from controller.models.domain.gafaelfawr import GafaelfawrUser
+
+from ..support.config import configure
+
+
+@pytest.mark.asyncio
+async def test_create_delete(
```

Comment:
Needs locking tests where you try to delete it at the same time that you're creating it. You can do that by having the pod start in pending and then start a delete and only afterwards change the pod state to running. I think there's a file server test like that.